serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc

```diff
@@ -0,0 +1,141 @@
+
+[IMPORTANT]
+====
+The **GVK** (Group, Version and Kind) and the **Name** values are **mandatory** and can't be empty, otherwise, the engine will throw a validation issue and the discovery will not be performed.  +
+The **namespace** is **optional**. In case of an empty value, the current namespace or context will be used.
+====
+
+== Configuration
+
+You can configure the Service Discovery with two **build time** properties, which are:
+
+* `quarkus.kogito.sw.discovery.enabled`: Enable the service discovery. Defaults to _true_.
+* `quarkus.kogito.kubernetes-client.okhttp.static.logger`: Weather or not to print the `http` communication between the client and the Kubernetes API. Defaults to _false_.
+
+
+== How does it work?
+
+During application startup, the service discovery is performed at the Quarkus _STATIC_INIT_ time. First, it scans +
+the Quarkus configuration values and searches for the Kubernetes URI pattern. Then if at least one is found, the engine +
+will parse the URI, query the Kubernetes API for the given resource, and override the given system property.
+
+Suppose we have an application that will consume a resource running on Kubernetes, this resource is a Knative Service +
+that exposes a function that can be discovered with the following URI:
+
+
+[source,shell]
+----
+org.kie.kogito.sw.knative.service=knative:v1/Service/serverless-workflow-greeting-quarkus/greeting-quarkus-cli
+----
+
+[NOTE]
+====
+Note that, the application property does not matter. The Service Discovery only cares about the application property value.
+====
+
+Once the application has started, you should be able to see in the logs the Service Discovery into action:
+
+[source,shell]
+----
+$ java -jar target/quarkus-app/quarkus-run.jar
+__  ____  __  _____   ___  __ ____  ______
+ --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
+ -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
+--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
+2022-08-23 12:21:04,360 DEBUG [org.kie.kog.add.qua.k8s.SWDiscoveryConfigSourceInterceptor] (main) Configuring k8s client...
+2022-08-23 12:21:05,245 INFO  [io.qua.sma.ope.run.OpenApiRecorder] (main) Default CORS properties will be used, please use 'quarkus.http.cors' properties instead
+2022-08-23 12:21:05,420 DEBUG [org.kie.kog.add.qua.k8s.SWDiscoveryConfigSourceInterceptor] (main) SUCCESS SWUrl parsed: knative:v1/Service/serverless-workflow-greeting-quarkus/greeting-quarkus-cli
+2022-08-23 12:21:06,043 INFO  [org.kie.kog.add.qua.k8s.SwKubeResource] (main) Connected to kubernetes cluster v1.23.4, current namespace is serverless-workflow-greeting-quarkus ---- Resource name is greeting-quarkus-cli
+2022-08-23 12:21:06,045 INFO  [org.kie.kog.add.qua.k8s.KnativeResourceDiscovery] (main) Trying to adapt kubernetes client to knative
+2022-08-23 12:21:06,316 INFO  [org.kie.kog.add.qua.k8s.KnativeResourceDiscovery] (main) Found Knative endpoint at http://greeting-quarkus-cli.serverless-workflow-greeting-quarkus.10.99.154.147.sslip.io
```

Comment:
shouldn't that be also a DEBUG info ?

Reply:
yes, both needs, update here and at the code.